src/solvers.jl

```diff
@@ -1,12 +1,20 @@
+using DiffEqBase
 using OrdinaryDiffEq
+using SteadyStateDiffEq
 using StochasticDiffEq
+using DiffEqJump
 
 import OrdinaryDiffEq: ODEProblem
+import SteadyStateDiffEq: SteadyStateProblem
 import StochasticDiffEq: SDEProblem
+import DiffEqJump: JumpProblem
 
 funcindex!(list, key, f, vals...) = list[key] = f(list[key],vals...)
 valueat(x::Number, t) = x
 valueat(f::Function, t) = f(t)
+transitionrate(S, T, k, rate, t) = exp(reduce((x,y)->x+log(S[y] <= 0 ? 0 : S[y]),
```

Comment:
Do we want to handle negatives like this here? I think that I would rather get an exception if we have a negative value for the state. That isn't supposed to be possible with these semantics for the petri nets.